packages/clerk-js/src/utils/injectedWeb3Providers.ts

```diff
@@ -29,19 +29,24 @@ interface EIP6963ProviderDetail {
 type EIP6963AnnounceProviderEvent = CustomEvent;
 type InjectedWeb3Provider = MetamaskWeb3Provider | CoinbaseWeb3Provider;
 
-class InjectedWeb3Providers {
+export class InjectedWeb3Providers {
   #providers: EIP6963ProviderDetail[] = [];
   #providerIdMap: Record<InjectedWeb3Provider, string> = {
     coinbase: 'Coinbase Wallet',
     metamask: 'MetaMask',
   } as const;
+  #hasSetupEventListeners = false;
 
   constructor() {
     if (typeof window === 'undefined') {
       return;
     }
-    window.addEventListener('eip6963:announceProvider', this.#onAnnouncement as EventListener);
-    window.dispatchEvent(new Event('eip6963:requestProvider'));
+
+    if (!this.#hasSetupEventListeners) {
+      window.addEventListener('eip6963:announceProvider', this.#onAnnouncement as EventListener);
+      window.dispatchEvent(new Event('eip6963:requestProvider'));
+      this.#hasSetupEventListeners = true;
+    }
```

Comment:
Make sure we only add a listener once, since we moved the instantiation [here](https://github.com/clerk/javascript/pull/4095/files#r1742721324)